core/controllers/resources.py

```diff
@@ -206,9 +212,54 @@ def put(self) -> None:
         logging.info(
             '[RELEASE COORDINATOR] %s saved promo-bar config property values: '
             '%s' % (self.user_id, promo_bar_message_value))
-        config_services.set_property(
-            self.user_id, 'promo_bar_enabled', promo_bar_enabled_value)
-        config_services.set_property(
-            self.user_id, 'promo_bar_message', promo_bar_message_value)
+        server_mode = (
+            'dev'
+            if constants.DEV_MODE
+            else 'prod'
+            if feconf.ENV_IS_OPPIA_ORG_PRODUCTION_SERVER
+            else 'test'
+        )
+        rules_for_promo_bar_enabled_value = [
+            platform_parameter_domain.PlatformParameterRule.from_dict({
+                'filters': [
+                    {
+                        'type': 'server_mode',
+                        'conditions': [['=', server_mode]]
+                    }
+                ],
+                'value_when_matched': promo_bar_enabled_value
+            })
+        ]
+        rules_for_promo_bar_message_value = [
+            platform_parameter_domain.PlatformParameterRule.from_dict({
+                'filters': [
+                    {
+                        'type': 'server_mode',
+                        'conditions': [['=', server_mode]]
+                    }
+                ],
+                'value_when_matched': promo_bar_message_value
+            })
+        ]
+        assert self.user_id is not None
+        try:
```

Comment:
Why have a try/except here? That just results in errors being "shadowed". Why would the platform parameter be "not found"?

If this is to catch a client error, you should try to prevent errors coming in from the client. The backend should just not accept bad data at the point of handler validation.

Reply:
Makes sense, I have removed the try/except. Also added the `assert` to check for the incoming value type.
Thanks